recipes/open62541/all/test_package/CMakeLists.txt

```diff
@@ -28,6 +28,22 @@ add_executable(${PROJECT_NAME}
     ${UA_TYPES_FOO_FLT_SOURCES}
     test_package.c
 )
+
+if (${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
+    target_compile_options(
+        ${PROJECT_NAME}
+        PRIVATE 
+        
+        -fsanitize=address
```

Comment:
Usually test packages are just testing for building/linking/making sure the package works. I don't know if there's a precedent for adding sanitizers to a test package.

Reply:
Do be honest, I do not know why this is necessary in the first place. If I do not add this linker option I get linker errors like these ones:
```
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_out_of_bounds [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_pointer_overflow [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_type_mismatch_v1 [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_nonnull_arg [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_divrem_overflow [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_sub_overflow [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_add_overflow [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_mul_overflow [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_shift_out_of_bounds [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __ubsan_handle_load_invalid_value [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __sanitizer_cov_trace_pc_guard [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __sanitizer_cov_trace_pc_guard_init [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __asan_register_globals [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __asan_unregister_globals [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __asan_init [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __asan_version_mismatch_check_v8 [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __asan_report_load_n [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __asan_loadN [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __asan_report_load1 [--no-allow-shlib-undefined]
ld: error: /home/conan/w/prod-v1/bsr/48966/cdbab/.conan/data/open62541/1.3.8/_/_/package/196f1131654c20996501278721a228f23b2dcd95/lib/libopen62541.so: undefined reference to __asan_report_load2 [--no-allow-shlib-undefined]
```

See https://c3i.jfrog.io/c3i/misc/logs/pr/20892/10-linux-clang/open62541/1.3.8//196f1131654c20996501278721a228f23b2dcd95-test.txt

Do you understand why these references are not found by the linker?

Reply:
Interesting. Do you know if 1.3.7/8 adds this to their build by default? If so, while I find it a little odd, it would mean that you need to add `-fsanitize=address` to the package's build and link flags in this version and up. We don't want consumers to have to have a weird link error when they try to use the package that they then have to fix, it should work out of the box.

Reply:
I think that would be reasonable. I imagine it's there for debugging of their code, so I don't know if it's that useful for use users of the package. That said, we could expose it as an option, which would conditionally set that variable and add those flags.

Reply:
Ok I exposed the option and changed the test_package back to the original.

Reply:
Looks like you also need to add a couple other things:
1. Pass the option from Conan to CMake through the CMake toolchain in `generate()`. Something like this existing option: `tc.variables["UA_ENABLE_SUBSCRIPTIONS"] = self.options.subscription != False`
2. If the option is enabled, modify the `package_info()` to add the sanitizer flags.

Reply:
Arg, yeah, that makes sense...

Reply:
Ok, the first part makes sense. But why should I enable the sanitizer flags in package_info? This would mean I have to duplicate this check from the cmake of the original open62541 Repo
https://github.com/open62541/open62541/blob/7837c1d6246fa78bba06d5e65ae82579867a1d5c/CMakeLists.txt#L652
and add all the sanitizer flags set in this line
https://github.com/open62541/open62541/blob/7837c1d6246fa78bba06d5e65ae82579867a1d5c/CMakeLists.txt#L657
.

That sounds odd to me.

Reply:
I just realized that `UA_ENABLE_SUBSCRIPTIONS` is not available in release 1.3.8. I will add a patch for this version that removes the sanitizers.